scripts/build_lockfiles.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+# prevent sourcing of this script, only allow execution
+$(return >/dev/null 2>&1)
+test "$?" -eq "0" && { echo "This script should only be executed." >&2; exit 1; }
+
+# exit on errors, undefined variables, ensure errors in pipes are not hidden
+set -Eeuo pipefail
+
+declare mydir
+mydir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+source "${mydir}/utils.sh"
+
+usage() {
+  msg
+  msg "Usage: $0 <pkg>"
+  msg
+  msg "\t<pkg> must be one of packages under packages/, e.g. 'hoprd'"
+  msg
+}
+
+# return early with help info when requested
+([ "${1:-}" = "-h" ] || [ "${1:-}" = "--help" ]) && { usage; exit 0; }
+
+if [ -z "${1:-}" ]; then
+  msg "Missing package name"
+  usage
+  exit 1
+fi
+
+declare package_name="${1}"
+
+declare build_dir="$(find_tmp_dir)/hopr_lockfile_generation"
+
+log "Creating temporary build directory ${build_dir} (will be removed)"
+mkdir "${build_dir}"
+
+declare package_dir="${mydir}/../packages/${package_name}"
+
+log "Install workspace package in temporary directory"
+
+declare package_npm_name=$(jq -r '.name' "${package_dir}/package.json")
+
+# Resolve workspace links by packing a NPM package
+yarn workspace "${package_npm_name}" pack
+
+cd "${build_dir}"
+
+# Decompress NPM package to get `package.json` file
+tar -xf "${package_dir}/package.tgz"
+
+# Turn yarn `resolutions` from workspace root into npm `overrides` for selected package
+jq -s '.[1] * (.[0].resolutions | { "overrides": . ,"resolutions": . })' "${mydir}/../package.json" "${build_dir}/package/package.json" > "${build_dir}/package.json"
+ 
+# # NPM package is now useless
+rm \
+  "${package_dir}/package.tgz" \
+  -R "${build_dir}/package"
```

Comment:
please split up into 2 rm invocations, makes it easier to comprehend

Reply:
changed